src/software/thunderscope/replay/proto_player.py

```diff
@@ -130,6 +134,91 @@ def is_log_entry_corrupt(log_entry: bytes, version: int) -> bool:
         except Exception:
             return True
 
+    def build_chunk_index(self, folder_path: str) -> dict[str:float]:
+        """Build the chunk index and store the index into the index file
+
+        Note:
+        ----
+        A chunk index entry is a key value pair [filename: start timestamp]
+         - Start timestamp: timestamp of the first log entry in the chunk
+         - Filename: replay chunk file name (%d.reply)
+
+        :param folder_path: folder containing all the replay files
+
+        :return: a dictionary for mapping replay filename to the start
+            timestamp of the first entry in the chunk.
+
+        """
+        chunk_indices: dict[str:float] = dict()
+        for chunk_name in self.sorted_chunks:
+            chunk_data = ProtoPlayer.load_replay_chunk(chunk_name, self.version)
+            if chunk_data:
+                start_timestamp, _, _ = ProtoPlayer.unpack_log_entry(
+                    chunk_data[0], self.version
+                )
+                chunk_indices[os.path.basename(chunk_name)] = start_timestamp
+        if chunk_indices:
+            with open(
+                os.path.join(folder_path, ProtoPlayer.CHUNK_INDEX_FILENAME), "w"
+            ) as index_file:
+                index_file.write(f"Generated on {time.time()}\n")
+                for key, value in chunk_indices.items():
+                    index_file.write(f"{value}, {key}\n")
+            logging.info("Created chunk index file successfully.")
```

Comment:
looks like you want to do serialization. 

An easier way to do this is to use the `pickle` library. https://docs.python.org/3/library/pickle.html 

Taken from a old code base of mine: 
loading an object:
```
def load_obj(path):
    try:
        return pickle.load(open(path, 'rb'))
    except Exception as e:
        return None
```

storing an object:
```
def save_obj(path, obj):
    with open(path, 'wb') as f:
        pickle.dump(obj, f)
```

I haven dig into this, but I think this would work in your case since you are only storing a dictionary with two serializable objects: float, and Proto?

Reply:
you should do error handling  for `save_obj` or something like that.

Reply:
I did think about using pickle.
Here is why I didn't choose pickle in the end:
1. Besides the dictionary, I also want to mantain a version number (or something like that, I am using a timestamp right now) in the index file. Although a version number is serializable, but it requires another data structure to wrap around.
2. I thought it might be helpful to keep this index file more human-readable by writing things to plain text? I am not sure about the context here, but I assumed someone would maually edit the log folder, and having this index might be handy.

But using pickle would definitely make everything easier in the sense of maintaining. Please let me know what you think.

Reply:
sure, I guess we could just write to a file with the current implementation.